### DISCUSSION TITLE
Need for a dev help on migrating from O1 with NHibernate to OCore with YesSQL

### DISCUSSION THREAD

**Discussion by CSurieux** [OrchardCMS/OrchardCore#15963] [Q&A]
[Sorry for cross posting, I first posted this in OrchardDoc repo but realized that it was the Orchard 1 Documentation, so I repeat my request here]

For all those as me who need to migrate their sites from Orchard 1 to OrchardCore, it would be very helpful to have some migration tips and recos for all the DB design and access.

I understand that Parts no more have dedicated DB tables (very nice) and are serialized in their document containers and that we have to create many indexes tables for each particular access.

My main questions would be :

How to replace the NH Part/Record schemas adopted for 1:N & N:N relations by classes and Indexes.
And where goes the old O1 int ContentItem Id ? What about all these string Ids in Core methods.

How to replace the various O1 queries by the new Core paradigms.

May be sticking to a server point of view without implying client Queries.

This help may exist in documentation but I have not been able to found it.

Thanks

**Comment by Skrypt** [comment]
There is no documentation about this that I know of. But I'm pretty sure everything is faisible by using only TSQL.

**Comment by sebastienros** [comment]
The recommendation is to generate an Orchard Core recipe file, like the ones for the Setup, that would contain your types and data. Then you don't need to care about YesSql. There are several recipe files showing how to create types and custom content items.

There is no migration for queries (I assume you mean SQL queries in this case), you'll have to understand how to create them in OC.

**Comment by Piedone** [comment]
Also see https://orcharddojo.net/blog/how-to-migrate-an-orchard-1-application-to-orchard-core.
